### PR TITLE
Allow URI queries without value

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: tests/uri.cpp [hunk 1]
```diff
@@ -288,3 +288,13 @@ BOOST_AUTO_TEST_CASE(test_host_port_without_schema)
     BOOST_CHECK_EQUAL( uri3.getHost(), "host" );
     BOOST_CHECK_EQUAL( uri3.getPort(), 12345 );
 }
+
+BOOST_AUTO_TEST_CASE(query_without_value)
+{
+    const servus::URI uri( "?foo,bar=foo,blubb" );
```
Comment (tribal-tec) [inline tests/uri.cpp]: what's happening with 'foo=' or '=foo'?

Comment (hernando) [inline tests/uri.cpp]: =foo will be ignored, which is reasonable. Another option will be throwing.
foo= should behave the same as foo, which is a key with an empty string as value. The code seems to do that but I wonder if std::string("foo").substr(3) works or crashes.

Comment (eile) [inline tests/uri.cpp]: added test for foo=

**Comment by tribal-tec** [comment]
+1

**Pull request merged**